File: C/ex03.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    float celsius, fahrenheit;
    char input[50];
    char *endptr;

    printf("Digite a temperatura em Celsius: ");
    fgets(input, sizeof(input), stdin);
    celsius = strtof(input, &endptr);

    if (input[0] == '\n' || *endptr != '\n') {
        printf("Entrada invalida. Por favor, digite uma temperatura valida em Celsius.\n");
        return 1;
    }

    fahrenheit = celsius * 1.8 + 32;

    printf("%.2f Celsius equivale a %.2f Fahrenheit\n", celsius, fahrenheit);

    return 0;
}

File: Haskell/ex03.hs
module Main where

import System.IO ()

main :: IO ()

main = do
    putStrLn "Digite a temperatura em Celsius: "
    celsius <- getLine
    let real_celsius = read celsius :: Float
    
    let fahrenheit = (real_celsius * 1.8) + 32

    putStrLn (show real_celsius ++ " °C em F é: " ++ show fahrenheit)